### PR TITLE
Decouple scda 

### PR DESCRIPTION
**Pull request by kartikeyakirar**
fixes https://github.com/insightsengineering/teal.transform/issues/133

### CODE REVIEW

**Comment by kartikeyakirar** [comment]
linking this issue; https://github.com/insightsengineering/teal.transform/issues/140

**Comment by gogonzo** [comment]
@kartikeyakirar you can remove all the tests which depends on `scda_dataset_connector`

File: tests/testthat/test-data_extract_module.R [hunk 1]
```diff
@@ -1,7 +1,5 @@
-library(scda)
-ADSL <- synthetic_cdisc_data("rcd_2022_06_27")$adsl # nolint
-ADLB <- synthetic_cdisc_data("rcd_2022_06_27")$adlb # nolint
-ADTTE <- synthetic_cdisc_data("rcd_2022_06_27")$adtte # nolint
+adlb <- rADLB # nolint
+adtte <- rADTTE # nolint
```
Comment (chlebowa) [inline tests/testthat/test-data_extract_module.R:2]: Let us keep the capitalization of data set names.
Also, that is what the `# nolint` is for.

Comment (gogonzo) [reply tests/testthat/test-data_extract_module.R:2]: Good point. Regarding uppercase dataset names in teal. In `teal.data` when using `cdisc_data` there is a check if `"ADSL"` exists, which is necessary for a join keys to work. It means we have this limitation in whole `teal` framework and we should stick to uppercase always when dealing with ADaM datasets. Other (then ADaM) datasets can be written in lowercase.

I think in the future we should make teal.data not being case-sensitive.

**Review by gogonzo — APPROVED** [review]
👍 Please wait with merge. We need to merge all packages from the same branch in the same time. We will be ready to merge them once this is approved by @lcd2yyz https://github.com/insightsengineering/osprey/pull/121

**Pull request merged**